test/api-helper/machine/AbstractSoftwareDeliveryMachine.test.ts

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright © 2018 Atomist, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Deferred } from "@atomist/automation-client/lib/internal/util/Deferred";
+import * as assert from "power-assert";
+import { resetRegistrableManager } from "../../../lib/api/machine/Registerable";
+import { TriggeredListenerRegistration } from "../../../lib/api/registration/TriggeredListenerRegistration";
+import { TestSoftwareDeliveryMachine } from "../TestSoftwareDeliveryMachine";
+
+describe("AbstractSoftwareDeliveryMachine", () => {
+
+    describe("addTriggeredListener", () => {
+
+        afterEach(() => {
+            resetRegistrableManager();
+        });
+
+        it("should register and schedule cron based trigger", () => {
+            let count = 0;
+            const job: TriggeredListenerRegistration = {
+                trigger: {
+                    cron: "*/2 * * * * *",
+                },
+                listener: async t => {
+                    assert(t.sdm);
+                    count = count + 1;
+                },
+            };
+            const sdm = new TestSoftwareDeliveryMachine("test");
+            sdm.addTriggeredListener(job);
+            return sdm.notifyStartupListeners()
+                .then(() => {
+                    const d = new Deferred<any>();
+                    setTimeout(() => {
+                        assert(count >= 2);
```

Comment:
Is this is currently only firing every even minute, how is it incrementing at least twice within ten seconds?

Reply:
The first element of the cron expression is for seconds. See https://github.com/kelektiv/node-cron#available-cron-patterns and https://github.com/kelektiv/node-cron/pull/392/files

Reply:
cc @ddgenome did you see this? Maybe it is not the best library if it isn't following crontab expressions. Although that was the same with Quartz for Java if I remember correctly.

Reply:
Wow